stock_picking_type_bypass_reservation/models/stock_picking_type.py

```diff
@@ -0,0 +1,20 @@
+# Copyright 2025 Moduon Team S.L.
+# License LGPL-3.0 or later (https://www.gnu.org/licenses/lgpl-3.0)
+
+from odoo import fields, models
+
+
+class StockPickingType(models.Model):
+    _inherit = "stock.picking.type"
+
+    bypass_reservation = fields.Boolean(
+        help="If checked, products from this Picking Type will always be reserved "
+        "even if you don't have enough stock (Available stock will be reserved as "
+        "usual).\nNote: Pickings will be reserved on Confirmation even if you don't "
+        "have checked the option",
+    )
+
+    def should_bypass_reservation(self):
```

Comment:
issue: this isn't meant to be called by ORM, so better make it private by default:

```suggestion
    def _should_bypass_reservation(self):
```

Reply:
just removed as unnecessary for the moment